webextension/background/analytics.js

```diff
@@ -307,15 +314,17 @@ this.analytics = (function() {
   }
 
   function fetchWatcher(request) {
-    catcher.watchPromise(
-      request.then(response => {
-        if (!response.ok) {
-          throw new Error(`Bad response from ${request.url}: ${response.status} ${response.statusText}`);
-        }
-        return response;
-      }),
-      true
-    );
+    request.then(response => {
+      if (response.status === 410) { // Gone
+        hasReturnedGone = true;
+        pendingEvents = [];
+      }
+      if (!response.ok) {
+        log.debug(`Error code in event response: ${response.status} ${response.statusText}`);
+      }
+    }).catch(error => {
+      log.debug(`Error event in response: ${error}`);
```

Comment:
Some users may not upgrade from 66 to 67 before the server is shut down in May, and `fetch` will throw if the server can't be found, so I think it'd be good to set `hasReturnedGone` inside this `catch`.

Reply:
My hope is that we can put in a placeholder site before the site goes down completely. Unfortunately we can't tell the difference between the site being offline, and the computer being offline. I've made a followup in  #5293 to discuss this.